src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/NuGetProjectFactory.cs

```diff
@@ -69,13 +64,6 @@ public sealed class NuGetProjectFactory
 
             await _threadingService.JoinableTaskFactory.SwitchToMainThreadAsync();
 
-            if (vsProjectAdapter.VsHierarchy != null &&
```

Comment:
This is very satisfying! :) 
It registers on every damn initialization UI delay.